tracer/contrib/redigo/tracedredis.go

```diff
@@ -0,0 +1,101 @@
+package tracedredis
+
+import (
+	"context"
+	"github.com/DataDog/dd-trace-go/tracer"
+	"github.com/DataDog/dd-trace-go/tracer/ext"
+	redis "github.com/garyburd/redigo/redis"
+	"net"
+	"net/url"
+	"strconv"
+	"strings"
+)
+
+type TracedConn struct {
+	redis.Conn
+	tracer  *tracer.Tracer
+	service string
+	network string
+	host    string
+	port    string
+}
+
+func TracedDial(service string, tracer *tracer.Tracer, network, address string, options ...redis.DialOption) (TracedConn, error) {
+	c, err := redis.Dial(network, address)
+	addr := strings.Split(address, ":")
+	host := addr[0]
+	port := addr[1]
+	tracer.SetServiceInfo(service, "redis", ext.AppTypeDB)
+	tc := TracedConn{c, tracer, service, network, host, port}
+	return tc, err
+}
+
+func (tc *TracedConn) SetService(service string) {
+	tc.tracer.SetServiceInfo(service, "redis", ext.AppTypeDB)
+	tc.service = service
+}
+
+func TracedDialURL(service string, tracer *tracer.Tracer, rawurl string, options ...redis.DialOption) (TracedConn, error) {
+	u, err := url.Parse(rawurl)
+	if err != nil {
+		return TracedConn{}, err
+	}
+	host, port, err := net.SplitHostPort(u.Host)
+	if err != nil {
+		host = u.Host
+		port = "6379"
+	}
+	if host == "" {
+		host = "localhost"
+	}
+	// Set in redis.DialUrl source code
+	network := "tcp"
+	c, err := redis.DialURL(rawurl, options...)
+	tc := TracedConn{c, tracer, service, network, host, port}
+	return tc, err
+}
+
+func (tc TracedConn) Do(commandName string, args ...interface{}) (reply interface{}, err error) {
+	ctx := context.Background()
+	ok := false
+	if len(args) > 0 {
+		ctx, ok = args[len(args)-1].(context.Context)
+		if ok {
+			args = args[:len(args)-1]
+		}
+	}
+
+	span := tc.tracer.NewChildSpanFromContext("redis.command", ctx)
+	defer func() {
```

Comment:
actually we have a more idiomatic way to do this: `defer span.FinishWithErr(err)` https://github.com/DataDog/dd-trace-go/blob/master/tracer/span.go#L171